src/lib.rs

```diff
@@ -7,21 +7,29 @@
 //! for each new connection in a source of new streams (such as a listening
 //! TCP or unix domain socket).
 
+mod compile_time_checks {
+    #[cfg(not(any(feature = "rustls", feature = "native-tls")))]
+    compile_error!("tls-listener requires either the `rustls` or `native-tls` feature");
+
+    #[cfg(all(feature = "rustls", feature = "native-tls"))]
+    compile_error!("The `rustls` and `native-tls` features in tls-listener are mutually exclusive");
+}
```

Comment:
These produce nice readable errors in the case that a user hasn't got their features setup right